scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py

```diff
@@ -36,77 +42,107 @@ def focus_run(self, sample_paths, banks, plot_output, instrument, rb_num, spectr
+                    dspacing_output_name = tof_output_name + "_dSpacing"
+                    if name == '1':
+                        df_kwarg = {"GroupingFileName": NORTH_BANK_CAL}
+                        region_calib = "engggui_calibration_bank_1"
+                    else:
+                        df_kwarg = {"GroupingFileName": SOUTH_BANK_CAL}
+                        region_calib = "engggui_calibration_bank_2"
+                    # need to clone these workspaces as they're altered in each run of focus
+                    sample_ws_clone = CloneWorkspace(sample_workspace)
+                    curves_ws_clone = CloneWorkspace(curves_workspace)
+                    self._run_focus(sample_ws_clone, tof_output_name, integration_workspace,
+                                    curves_ws_clone, df_kwarg, full_calib_workspace, region_calib)
+                    workspaces_for_run.append(tof_output_name)
+                    # Save the output to the file system.
+                    self._save_output(instrument, sample_path, name, tof_output_name, rb_num)
+                    self._save_output(instrument, sample_path, name, dspacing_output_name, rb_num)
+                    DeleteWorkspace(sample_ws_clone)
+                    DeleteWorkspace(curves_ws_clone)
+                output_workspaces.append(workspaces_for_run)
                 self._output_sample_logs(instrument, run_no, sample_workspace, rb_num)
+
         # Plot the output
         if plot_output:
             for ws_names in output_workspaces:
                 self._plot_focused_workspaces(ws_names)
 
     @staticmethod
     def _run_focus(input_workspace,
-                   output_workspace,
+                   tof_output_name,
                    vanadium_integration_ws,
                    vanadium_curves_ws,
-                   bank,
-                   full_calib_ws=None,
-                   spectrum_numbers=None):
-        kwargs = {
-            "InputWorkspace": input_workspace,
-            "OutputWorkspace": output_workspace,
-            "VanIntegrationWorkspace": vanadium_integration_ws,
-            "VanCurvesWorkspace": vanadium_curves_ws
-        }
-        if full_calib_ws is not None:
-            kwargs["DetectorPositions"] = full_calib_ws
-
-        if bank is None:
-            kwargs["SpectrumNumbers"] = spectrum_numbers
-        else:
-            kwargs["Bank"] = bank
-
-        try:
-            return EnggFocus(**kwargs)
-        except RuntimeError as e:
-            logger.error(
-                "Error in focusing, Could not run the EnggFocus algorithm successfully for bank "
-                + str(bank) + ". Error Description: " + str(e))
-            raise RuntimeError()
+                   df_kwarg,
+                   full_calib,
+                   region_calib):
+        NormaliseByCurrent(InputWorkspace=input_workspace, OutputWorkspace=input_workspace)
```

Comment:
In #31213 I added a check to see if proton charge was finite before calling Normalise by current - I think we might need a similar check in this new workflow (though I haven't tested it with zero proton charge data yet).